internal/tools/aws/eks.go

```diff
@@ -81,325 +131,439 @@ func (a *Client) InstallEKSEBSAddon(cluster *model.Cluster) error {
+		return nil, errors.New("EKS NodeGroup creation failed")
+	}
 
-func (a *Client) listNodeGroups(ctx context.Context, clusterName string) ([]*eksTypes.Nodegroup, error) {
-	listNgInput := eks.ListNodegroupsInput{
-		ClusterName: aws.String(clusterName),
+	if nodeGroup.Status == eksTypes.NodegroupStatusActive {
+		return nodeGroup, nil
 	}
 
-	nodeGroups := make([]*eksTypes.Nodegroup, 0)
+	return nil, nil
+}
+
+// WaitForActiveEKSCluster waits for EKS cluster to be ready.
+func (c *Client) WaitForActiveEKSCluster(clusterName string, timeout int) (*eksTypes.Cluster, error) {
+	timer := time.NewTimer(time.Duration(timeout) * time.Second)
+	defer timer.Stop()
+
 	for {
-		ngListOut, err := a.Service().eks.ListNodegroups(ctx, &listNgInput)
-		if err != nil {
-			// If cluster does not exist anymore, listing node groups will
-			// fail with ResourceNotFoundException.
-			if IsErrorResourceNotFound(err) {
-				return nodeGroups, nil
+		select {
+		case <-timer.C:
+			return nil, errors.New("timed out waiting for EKS cluster to become active")
+		default:
+			eksCluster, err := c.GetActiveEKSCluster(clusterName)
+			if err != nil {
+				return nil, errors.Wrap(err, "failed to check if EKS cluster is active")
+			}
+			if eksCluster != nil {
+				return eksCluster, nil
 			}
-			return nil, errors.Wrap(err, "failed to list node groups for the cluster")
+
+			time.Sleep(5 * time.Second)
 		}
+	}
+}
 
-		for _, ng := range ngListOut.Nodegroups {
-			ngInput := eks.DescribeNodegroupInput{
-				ClusterName:   aws.String(clusterName),
-				NodegroupName: aws.String(ng),
-			}
-			out, err := a.Service().eks.DescribeNodegroup(ctx, &ngInput)
+func (c *Client) WaitForActiveEKSNodeGroup(clusterName, workerName string, timeout int) (*eksTypes.Nodegroup, error) {
+	timer := time.NewTimer(time.Duration(timeout) * time.Second)
+	defer timer.Stop()
+
+	for {
+		select {
+		case <-timer.C:
+			return nil, errors.New("timed out waiting for EKS NodeGroup to become active")
+		default:
+			nodeGroup, err := c.GetActiveEKSNodeGroup(clusterName, workerName)
 			if err != nil {
-				return nil, errors.Wrapf(err, "failed to describe node group %q", ng)
+				return nil, errors.Wrap(err, "failed to check if EKS NodeGroup is active")
+			}
+			if nodeGroup != nil {
+				return nodeGroup, nil
 			}
-			nodeGroups = append(nodeGroups, out.Nodegroup)
+
+			time.Sleep(5 * time.Second)
 		}
+	}
+}
 
-		if ngListOut.NextToken == nil {
-			break
+func (c *Client) WaitForEKSNodeGroupToBeDeleted(clusterName, workerName string, timeout int) error {
+	timer := time.NewTimer(time.Duration(timeout) * time.Second)
+	defer timer.Stop()
+
+	for {
+		select {
+		case <-timer.C:
+			return errors.New("timed out waiting for EKS NodeGroup to become ready")
+		default:
+			nodeGroup, err := c.getEKSNodeGroup(clusterName, workerName)
+			if err != nil {
+				return errors.Wrap(err, "failed to describe NodeGroup")
+			}
+			if nodeGroup == nil {
+				return nil
+			}
+
+			time.Sleep(5 * time.Second)
```

Comment:
nit: I'd rather use `time.Ticker`, but that's a personal preference and I don't think it will matter much in this scenario.

Reply:
That is a good option.